### PR TITLE
Milestone 2: Feature user interface

### PR DESCRIPTION
**Pull request by Aduda-Boaz**
This PR is for milestone 2. The aim of the milestone is to create the user interface for the tic tac toe game, to be played on the terminal. The project is a collaborative exercise, built with:-

1. Code Editor
2. Terminal (Bash)
3. Git
4. Ruby

We have refactored our codes into different files to achieve our desired user interface.
This project is key to our learning at Microverse and we believe in any feedback to be vital in improving our learning experience.
We look forward to any feedback. Thank you in advance for your time.

### CODE REVIEW

**Review by Berabjesus — CHANGES_REQUESTED** [review]
# Status: CHANGES REQUIRED :warning: :construction: 
Hi @Aduda-Boaz  and @joshuaivie  👋. Awesome start on the project.
### However, you do have a few changes to carry out before we proceed so please take a look at the things you have to work on and their respective review comments below.

- [x] game informs a player if the selected move is a winning move (without computing the winning move). [Example screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/game_win%2059.png)
- [x] game informs player if the selected move is a draw move (without computing draw). [Example screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/game_tie.png)
- Currently, the game is just displaying a thank you message. Please implement some conditions to show a WIN and DRAW message.
- [OPTIONAL NOTE] You didn't have to create the logic in the lib directory just yet as you will do so in the next milestone.

#### **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

**Comment by joshuaoivie** [comment]
Dear Code Reviewer,

The following changes have been made in line with the previous review.

- Add #game_over method
- Randomly select decide a win or draw
- In the case of a win, randomly decide the winner.

We're looking forward to your objective review.

Warm Regards,
Josh

**Review by MkrtichSargsyan — CHANGES_REQUESTED** [review]
## _Status:_  Changes Requested :recycle:

### Hello Guys


Good job with the changes you made after the last review :+1:,
However, there are some changes that you need to make in order to get your project approved

- [x] user input should be validated so it's not empty. ( users should not be able to create an empty username)
  Kindly add a function that will check if the user input is not empty, otherwise ask the user to enter a valid user name )

- [x] when one of the players makes an invalid move the player turn is changed, please make sure that the same player should make the move 

err


## Optional

In the Game class you have ```attr_reader :player_one, :player_two, :active_player```, but you are not using those accessors. If you don't plan to use them, make sure to remove them.


Please, make the REQUIRED changes and submit your request again! :grinning:
Good luck. 

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+require_relative './modules/gameutils'
+require_relative './player'
+require_relative './board'
+
+class Game
+  include GameUtils
+  attr_reader :player_one, :player_two, :active_player, :max_moves, :moves_made
```
Comment (MkrtichSargsyan) [inline lib/game.rb:7]: You are not using some of these accessors (:player_one, :player_two, :active_player,). If you don't plan to use them, make sure to remove them.

Comment (joshuaoivie) [reply lib/game.rb:7]: We're using these accessors in the project.

Comment (MkrtichSargsyan) [reply lib/game.rb:7]: I checked that, it seems you are not using them. But I will check it again.

**Review by mke2111 — APPROVED** [review]
# 😄 🥇  STATUS: APPROVED  🥇 👏🏽 

## Hey Happy Coders, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval. 👍🏽 

### **_Well Done_** 💯 

    1. The user interface is as expected
    2. The game runs just as expected from the given instructions
    3. The linter checks run just fine.

### You may now merge to the develop branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 

<hr>

### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Comment by joshuaoivie** [comment]
Thank you so much for your review @mke2111!

**Pull request merged**